bigchaindb/pipelines/block.py

```diff
@@ -55,19 +55,20 @@ def validate_tx(self, tx):
             tx (dict): the transaction to validate.
 
         Returns:
-            The transaction if valid, ``None`` otherwise.
+            The transaction (Transaction) if valid, ``None`` otherwise.
         """
-        if self.bigchain.transaction_exists(tx['id']):
+        tx = Transaction.from_dict(tx)
+        if self.bigchain.transaction_exists(tx.id):
```

Comment:
@rhsimplex: This is new logic that was added in this commit: https://github.com/bigchaindb/bigchaindb/pull/641/commits/c33d1230f2c8fdca2688c8a45ad94458cacce92c

Would it make sense to put it in: https://github.com/bigchaindb/bigchaindb/pull/641/files#diff-9f1ab5b5daf1b3072e2123b6939bfb07R13 ?